pkgs/by-name/th/thorium-reader/package.nix

```diff
@@ -0,0 +1,84 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  appimageTools,
+  undmg ? null,
+}:
+
+let
+  pname = "thorium-reader";
+  version = "3.2.2";
+  srcs = {
+    x86_64-linux = fetchurl {
+      url = "https://github.com/edrlab/thorium-reader/releases/download/v${version}/Thorium-${version}.AppImage";
+      sha256 = "1786sk811pbxr0mzgcyn49hld15m7asfy8a7fx1ld4xvkl236iq6";
+    };
+    aarch64-linux = fetchurl {
+      url = "https://github.com/edrlab/thorium-reader/releases/download/v${version}/Thorium-${version}-arm64.AppImage";
+      sha256 = "0hzmh2km7ykm34asb7dhyvmmchn7j87vjdskz90m7jr3i70ixn32";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://github.com/edrlab/thorium-reader/releases/download/v${version}/Thorium-${version}.dmg";
+      sha256 = "055rm57vx8fcq7vqzkkya7a857ri6ikkb98im1qwv2pymzkglzmb";
+    };
+    aarch64-darwin = fetchurl {
+      url = "https://github.com/edrlab/thorium-reader/releases/download/v${version}/Thorium-${version}-arm64.dmg";
+      sha256 = "0b7nckfql9w5xyndp2z3b32jnw31dll12xz3v74lb9ngj60pbv7x";
+    };
+  };
+  meta = {
+    description = "Thorium Reader - EPUB reader";
+    homepage = "https://github.com/edrlab/thorium-reader";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ YodaDaCoda ];
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    mainProgram = "thorium-reader";
+  };
```

Comment:
It is recommended to add `sourceProvenance = with sourceTypes; [ binaryBytecode ];` here if you really want to use the binary source.